client.go

```diff
@@ -110,6 +119,80 @@ func chopContentType(contentType string) string {
 	return strings.Split(contentType, ";")[0]
 }
 
+// Subscribe function subscribes for events form the StreamClient TopicName after the given offset. An offset of zero should be
+// provided to read from the beginning. The provided EventHandler function will be called for each value.
+// To deal with errors while reading messages, an error handler function should also be provided.
+//
+// The function returns a context.CancelFunc which may be called for cancelling the subscription.
+func (lc *StreamClient) Subscribe(ctx context.Context, group string, offset uint64, f EventHandler, e EventErrHandler) (context.CancelFunc, error) {
+	subContext, cancel := context.WithCancel(ctx)
+	request := liiklus.SubscribeRequest{
+		Topic:                lc.TopicName,
+		Group:                group,
+		AutoOffsetReset:      liiklus.SubscribeRequest_EARLIEST,
+	}
+	subscribedClient, err := lc.client.Subscribe(subContext, &request)
+	if err != nil {
+		return cancel, err
```

Comment:
The cancelFunc returned should be nil in case of error

Reply:
That is how it was, changed because `go vet` complained.

Reply:
Curious to see what `go vet` has to say in detail for this particular case. The general rule is to return zero values in case of errors.

Reply:
This is what go vet said:
```
./client.go:129:2: the cancel function is not used on all paths (possible context leak)
./client.go:137:3: this return statement may be reached without using the cancel var defined on line 129
```

Reply:
OK, I think then that we should rather make sure that this function cancels the context itself, not return the cancelFunc (which the user may happily forget to use)

Reply:
We return cancelFunc so that users can call it to cancel the "subscription". 
It is ok if the users do not call cancelFunc, since all we are doing is wrapping the user passed context. So, when the user passed context is terminated, the context we created will terminate too.